Test message db

yajaneya opened the pull request:
Добавил тест добавления и нахождения сообщений в БД

https://trello.com/c/z21fv7IJ/116-%D1%81%D0%BE%D0%B7%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5-%D1%82%D0%B5%D1%81%D1%82%D0%B0-%D0%B4%D0%BE%D0%B1%D0%B0%D0%B2%D0%BB%D0%B5%D0%BD%D0%B8%D1%8F-%D0%B8-%D0%BD%D0%B0%D1%85%D0%BE%D0%B6%D0%B4%D0%B5%D0%BD%D0%B8%D1%8F-%D1%81%D0%BE%D0%BE%D0%B1%D1%89%D0%B5%D0%BD%D0%B8%D0%B9-%D0%B2-%D0%B1%D0%B4

yajaneya committed:
Merge pull request #5 from siblis/develop

Develop

yajaneya committed:
Добавил тест метода по добавлению сообщений в БД:
public void addMessage(Message message) из class DataBaseService

PaLoMaster commented:
Думаю, что надо чуть поправить addMessageTest (кстати, и название метода тоже):
1. Сделать список сообщений.
2. Добавить сообщения из списка в БД (по одному).
3. Проверить, что список сообщений из getChat соответствует списку сообщений пункта 1.

PaLoMaster commented:
И игноры надо убрать)

yajaneya committed:
- убрал закомментированные Ignore
- сделал пакетное тестирование: создается список сообщений; добавляется в БД; получается список сообщений с БД; сверяются изначальный список и полученный
- добавил методы equals() и hashCode() в класс Message для возможности осуществления сверки сообщений.

yajaneya committed:
Переименовал имена тестовых методов в ...Test()

PaLoMaster commented on src/test/java/ru/geekbrains/pocket/messenger/database/dao/DataBaseServiceTest.java:
```diff
@@ -3,50 +3,87 @@
+
+        for (int i=0; i<messages.size(); i++) {
+            Assert.assertEquals(messages.get(i), messages1.get(i));
+        }
```
`Assert.assertEquals(messages, messages1)` вроде должно отработать корректно...

PaLoMaster commented on src/test/java/ru/geekbrains/pocket/messenger/database/dao/DataBaseServiceTest.java:
```diff
@@ -3,50 +3,87 @@
     @After
     public void tearDown() throws Exception {
-            HibernateUtil.deleteDBFile();
+            HibernateUtil.deleteDBFile(); // удалятеся тестовая БД
```
ОЧеПятка (удаляТЕся)? :)

PaLoMaster reviewed (commented):
Можно улучшить...? :)

yajaneya committed:
Внес улучшение - сделал сравнение списков.
Исправил опечатку
